packages/wonder-blocks-dropdown/components/dropdown-core.js

```diff
@@ -65,22 +79,59 @@ export default class DropdownCore extends React.Component<DropdownCoreProps> {
     }
 
     componentDidMount() {
+        this.updateEventListeners();
+    }
+
+    componentDidUpdate(prevProps: DropdownCoreProps) {
+        if (prevProps.open !== this.props.open) {
+            this.updateEventListeners();
+        }
+
+        // If the menu just changed from closed to open, and if we haven't
+        // found the correct menuWidth yet. Only applies to right aligned.
+        if (this.props.alignment === "right" && !prevProps.open && this.props.open && !this.menuWidth) {
+            // There is a timeout because if we call getBoundingClientRect
+            // right away, it returns the value of the document width instead
+            // of the width of the just-mounted component, the menu.
+            requestAnimationFrame(() => {
+                if (this.menuElement) {
+                    const rect = this.menuElement.getBoundingClientRect();
+                    // This update forcing would only happen once, because now
+                    // this.menuWidth is set
+                    this.menuWidth = rect.width;
+                    this.forceUpdate();
+                }
+            });
+        }
+    }
+
+    componentWillUnmount() {
+        this.removeEventListeners();
+    }
+
+    updateEventListeners() {
+        if (this.props.open) {
+            this.addEventListeners();
+        } else {
+            this.removeEventListeners();
+        }
+    }
+
+    addEventListeners() {
         document.addEventListener("mouseup", this._handleInteract);
         document.addEventListener("touchend", this._handleInteract);
         document.addEventListener("keyup", this._handleKeyup);
     }
 
-    componentWillUnmount() {
+    removeEventListeners() {
         document.removeEventListener("mouseup", this._handleInteract);
         document.removeEventListener("touchend", this._handleInteract);
         document.removeEventListener("keyup", this._handleKeyup);
     }
 
     _handleInteract = (event: {target: any}) => {
-        if (this.node && this.node.contains(event.target)) {
-            return;
-        }
-        if (this.props.open) {
+        const target: Node = event.target;
```

Comment:
Here, too, with props. I roughly keep the rule of "if more than one prop is used, then do destructuring to get the variables".